homeassistant/components/octoprint/sensor.py

```diff
@@ -4,64 +4,61 @@
 
 from pyoctoprintapi import OctoprintJobInfo, OctoprintPrinterInfo
 
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import (
+    CONF_NAME,
     DEVICE_CLASS_TEMPERATURE,
     DEVICE_CLASS_TIMESTAMP,
     PERCENTAGE,
     TEMP_CELSIUS,
 )
+from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity import Entity
 from homeassistant.helpers.update_coordinator import (
     CoordinatorEntity,
     DataUpdateCoordinator,
 )
 
-from . import CONF_BED, CONF_NUMBER_OF_TOOLS, DOMAIN as COMPONENT_DOMAIN
+from . import DOMAIN as COMPONENT_DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Set up the available OctoPrint sensors."""
-    if discovery_info is None:
-        return
-
-    name = discovery_info["name"]
-    base_url = discovery_info["base_url"]
-    monitored_conditions = discovery_info["sensors"]
-    number_of_tools = discovery_info[CONF_NUMBER_OF_TOOLS]
-    bed = discovery_info[CONF_BED]
-    coordinator: DataUpdateCoordinator = hass.data[COMPONENT_DOMAIN][base_url]
-    tools = []
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: ConfigEntry, async_add_entities
+):
+    """Set up the available OctoPrint binary sensors."""
+    coordinator: DataUpdateCoordinator = hass.data[COMPONENT_DOMAIN][
+        config_entry.entry_id
+    ]["coordinator"]
+    device_id: str = hass.data[COMPONENT_DOMAIN][config_entry.entry_id]["device_id"]
+    entities = []
+    sensor_name = config_entry.data[CONF_NAME]
     if coordinator.data["printer"]:
-        tools = [tool.name for tool in coordinator.data["printer"].temperatures]
+        try:
+            printer_info = coordinator.data["printer"]
+            types = ["actual", "target"]
+            for tool in printer_info.temperatures:
+                for temp_type in types:
+                    entities.append(
+                        OctoPrintTemperatureSensor(
+                            coordinator, sensor_name, tool.name, temp_type, device_id
+                        )
+                    )
+        except BaseException as ex:  # pylint: disable=broad-except
```

Comment:
What is doing I/O here?

Reply:
Removed, left over from when it did an API call